n_helpers.c

```diff
@@ -150,60 +287,87 @@ const char *NoteBinaryEncode(const uint8_t *inBuf, uint32_t inLen,
+             resulting encoded data is guaranteed to fit in the provided buffer.
+          2. When data is transmitted to the Notecard, this API can be used to
+             verify whether or not unencoded data of a given length will fit in
+             the provided buffer after encoding.
+
+  @param  size The size of the fixed-size buffer.
+
+  @returns  The max length of unencoded data certain to fit in the fixed-size
+            buffer, after being encoded.
+ */
 /**************************************************************************/
-uint32_t NoteBinaryEncodedLength(const uint8_t *buf, uint32_t len)
+uint32_t NoteBinaryMaxDecodedLength(uint32_t bufferSize)
 {
-    return cobsEncodedLength(buf, len);
+    return cobsGuaranteedFit(bufferSize);
 }
 
 //**************************************************************************/
 /*!
-  @brief  Given an input buffer's length, compute the maximum output buffer size
-          needed to binary encode the input buffer.
-  @param  len The length of the input buffer.
+  @brief  Compute the maximum buffer size needed to encode
+          any unencoded buffer of the given length.
+
+  @param  unencodedLength The length of an unencoded buffer.
+
   @returns  The max required buffer size to hold the encoded data.
-*/
+ */
 /**************************************************************************/
-uint32_t NoteBinaryEncodedMaxLength(uint32_t len)
+uint32_t NoteBinaryMaxEncodedLength(uint32_t unencodedLength)
 {
-    return cobsEncodedMaxLength(len);
+    return cobsEncodedMaxLength(unencodedLength);
 }
 
 //**************************************************************************/
 /*!
   @brief  Receive a large binary object from the Notecard's binary buffer
+
   @param  buffer A buffer to hold the binary object
   @param  bufLen The total length of the provided buffer
-  @param  dataLen An out parameter to hold the length of the decoded data in the
-                  output buffer.
+  @param  decodedOffset The offset to the decoded binary data already residing
+                        on the Notecard
+  @param  decodedLen [in/out] The length of the decoded data to fetch from the
+                     Notecard. If you wish to fetch the entire buffer from the
+                     given offset, set this value to `NOTE_C_BINARY_RX_ALL`.
+                     This parameter will return the bytes actually received from
+                     the Notecard.
+
   @returns  NULL on success, else an error string pointer.
-  @note  Buffers are decoded in place. The original contents of the buffer
-         will be modified.
-  @note  To determine the necessary size for the buffer, use
-         NoteBinaryRequiredRxBuffer.
-*/
+
+  @note  The buffer must be large enough to hold the encoded value of the
+         data store contents from the requested offset for the specified length.
+  @note  To determine the necessary buffer size for a given data length, use
```

Comment:
This note is sort of just a continuation of the previous one. I'd just consolidate it into one note. I'd also explain that the +1 below if or the newline. Also, `NoteBinaryMaxEncodedLength` takes a parameter, so I'd change `NoteBinaryMaxEncodedLength() + 1` to `NoteBinaryMaxEncodedLength(decodedLen) + 1`.